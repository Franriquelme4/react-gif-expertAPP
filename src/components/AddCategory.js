import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
        console.log('');
    }
    const handleSubmit=(e)=>{
        e.preventDefault();
        console.log(inputValue);
        if (inputValue) {
            setCategories(cat => [inputValue,...cat]);
            setInputValue('');
        }
    }

    return ( 
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}

                />
        </form>
    
    );
}
 AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
 }
export default AddCategory;