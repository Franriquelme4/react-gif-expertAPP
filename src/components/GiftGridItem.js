import React from 'react';
import PropTypes from 'prop-types';
export const GiftGridItem = ({title,url}) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
  )
}

GiftGridItem.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
}

 /*
  installar enxyme
  1) enzyme
  2) enzyme json
  3) debe mostrar el componente correctamente
    *shallow
    *wrapper
    *wrapper . toMatchSnapshot
 
 */