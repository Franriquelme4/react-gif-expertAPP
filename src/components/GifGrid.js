
import React, { Fragment } from "react";
import '@testing-library/jest-dom'
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GiftGridItem } from "./GiftGridItem";
import PropTypes from 'prop-types';
export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  // se renombra

  return (
    <Fragment>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      <div className="img-container">
        {images.map((image) => (
          <GiftGridItem key={image.id} {...image} />
        ))}
      </div>
    </Fragment>
  );
};
GifGrid.propTypes={
  category:PropTypes.string.isRequired
}