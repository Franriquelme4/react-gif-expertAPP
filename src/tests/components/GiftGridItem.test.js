import '@testing-library/jest-dom';
import React from 'react';
//import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import { GiftGridItem } from '../../components/GiftGridItem';
 '../../GiftExpertApp';
describe('SE prueba la carga de los componentes',()=>{
    const title = 'titulo';
    const url = 'http://localhost/algo.jpg';
    const wrapper =  shallow(<GiftGridItem title={title} url={url}/>);

    test('Se prueba la carga de los componentes', () => { 
        expect(wrapper).toMatchSnapshot();
     })
     test('debe de tener un parrafo con el title',  ()=>{
         const p= wrapper.find('p');
         expect(p.text().trim()).toBe(title);
     } )


     test('Debe tener la imagen igual al props', () => { 
        const img =  wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
      })

      test('Debe tener animate_fadeIn', () => { 
          const div =  wrapper.find('div');
        console.log(div.prop('className'));
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
       })


GiftGridItem
})