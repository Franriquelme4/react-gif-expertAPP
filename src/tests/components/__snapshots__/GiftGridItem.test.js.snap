// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SE prueba la carga de los componentes Se prueba la carga de los componentes 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
